src-electron/main-process/modules/wallet-rpc.js

```diff
@@ -1024,13 +1044,112 @@ export class WalletRPC {
     }
   }
 
+  /*
+  Get the LNS records cached in this wallet. 
+  */
+  async lnsKnownNames() {
+    try {
+      let params = {
+        decrypt: true,
+        include_expired: false
+      };
+
+      let data = await this.sendRPC("lns_known_names", params);
+
+      if (data.result && data.result.known_names) {
+        return data.result.known_names;
+      } else {
+        console.debug("No known names found");
+        return [];
+      }
+    } catch (e) {
+      console.debug("There was an error getting known records: " + e);
+      return [];
+    }
+  }
+
+  /*
+  Renews an LNS (Lokinet) mapping, since they can expire
+  type can be:
+  lokinet_1y, lokinet_2y, lokinet_5y, lokinet_10y
+  */
+  lnsRenewMapping(password, type, name) {
+    let _name = name.trim().toLowerCase();
+
+    // the RPC accepts names with the .loki already appeneded only
+    // can be lokinet_1y, lokinet_2y, lokinet_5y, lokinet_10y
+    if (type.startsWith("lokinet")) {
+      _name = _name + ".loki";
+    }
+
+    crypto.pbkdf2(
```

Comment:
We should add a check here to return an error if type is not the ones listed. Would help catch bugs in the future

Reply:
I can put in a log, but technically the user should never get to here with the wrong type, since they are limited only to the drop down. If the type is wrong, it'll show as "invalid type" from the RPC anyway (which will be displayed to the user)